### PR TITLE
Increase `max_memory_threshold_mb` values for `TestNLULeakManyEpochs` and `TestCoreLeakManyEpochs` tests

### LINKED ISSUE #8654: Flakey CI test: Memory Leak Assertion Error
**Error information**:
The test fails because of AssertionError, for example:
```
 assert max_memory_usage <  self.max_memory_threshold_mb
E       assert 1923.11328125 < 1600
E        +  where 1600 = <tests.test_memory_leak.TestNLULeakManyEpochs object at 0x7f6291693370>.max_memory_threshold_mb

tests/test_memory_leak.py:96: AssertionError
```

Job run URLs:
https://github.com/RasaHQ/rasa/runs/2409914416
https://github.com/RasaHQ/rasa/runs/2455744858
https://github.com/RasaHQ/rasa/runs/2507865963 
https://github.com/RasaHQ/rasa/runs/2527416813
https://github.com/RasaHQ/rasa/runs/2517392398

Definition of Done:
- [x] determine if the threshold is wrong or if there is an actual memory leak.
- [ ] record what the expected memory fluctuation is on the CI runner. 
- [ ] set the correct threshold, or create issue to fix leak.

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- fix #8654 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by alwx** [comment]
Running a bunch of tests today to figure out if the memory consumption is consistent

**Comment by alwx** [comment]
I ran a bunch of tests, and the results for them were pretty consistent — no memory leaks were detected. So I decided to increase the memory threshold.


```
________________ TestNLULeakManyEpochs.test_for_memory_leak[0] _________________
tests/test_memory_leak.py:105: in test_for_memory_leak
    assert max_memory_usage < self.max_memory_threshold_mb
E   assert 3490.3046875 < 1600
E    +  where 1600 = <tests.test_memory_leak.TestNLULeakManyEpochs object at 0x7fe15a383370>.max_memory_threshold_mb
________________ TestNLULeakManyEpochs.test_for_memory_leak[1] _________________
tests/test_memory_leak.py:105: in test_for_memory_leak
    assert max_memory_usage < self.max_memory_threshold_mb
E   assert 3642.79296875 < 1600
E    +  where 1600 = <tests.test_memory_leak.TestNLULeakManyEpochs object at 0x7fe15a38bbb0>.max_memory_threshold_mb
________________ TestNLULeakManyEpochs.test_for_memory_leak[2] _________________
tests/test_memory_leak.py:105: in test_for_memory_leak
    assert max_memory_usage < self.max_memory_threshold_mb
E   assert 3455.61328125 < 1600
E    +  where 1600 = <tests.test_memory_leak.TestNLULeakManyEpochs object at 0x7fe15a399eb0>.max_memory_threshold_mb
________________ TestNLULeakManyEpochs.test_for_memory_leak[3] _________________
tests/test_memory_leak.py:105: in test_for_memory_leak
    assert max_memory_usage < self.max_memory_threshold_mb
E   assert 3442.015625 < 1600
E    +  where 1600 = <tests.test_memory_leak.TestNLULeakManyEpochs object at 0x7fe16a85f040>.max_memory_threshold_mb
________________ TestNLULeakManyEpochs.test_for_memory_leak[4] _________________
tests/test_memory_leak.py:105: in test_for_memory_leak
    assert max_memory_usage < self.max_memory_threshold_mb
E   assert 3506.0546875 < 1600
E    +  where 1600 = <tests.test_memory_leak.TestNLULeakManyEpochs object at 0x7fe16a815400>.max_memory_threshold_mb
```

File: tests/test_memory_leak.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def epochs(self) -> int:
 
     @property
     def max_memory_threshold_mb(self) -> float:
-        return 1600
+        return 2200
```
Comment (ancalita) [inline tests/test_memory_leak.py:130]: How did you decide on the new float thresholds?

Comment (alwx) [reply tests/test_memory_leak.py:130]: Just increased them based on the information from several CI runs. + did the memory leak investigation locally (to make sure there is no memory leak)

**Review by ancalita — APPROVED** [review]
Awesome 🚀

**Pull request merged**